distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -323,26 +323,55 @@ Text.match self pattern=".*" case_sensitivity=Case_Sensitivity.Sensitive =
-                Case_Sensitivity.Sensitive ->
-                    Text_Utils.span_of_all self delimiter
-                Case_Sensitivity.Insensitive locale ->
-                    Text_Utils.span_of_all_case_insensitive self delimiter locale.java_locale
-            Vector.new delimiters.length+1 i->
-                start = if i == 0 then 0 else
-                    delimiters.at i-1 . codeunit_end
-                end = if i == delimiters.length then (Text_Utils.char_length self) else
-                    delimiters.at i . codeunit_start
-                Text_Utils.substring self start end
-        True ->
-            Helpers.regex_assume_default_locale case_sensitivity <|
-                case_insensitive = case_sensitivity.is_case_insensitive_in_memory
-                compiled_pattern = Regex_2.compile delimiter case_insensitive=case_insensitive
-                compiled_pattern.split self only_first
+
+   > Example
+     Split with a vector of strings.
+
+         'azbzczdzezfzg'.split ['b', 'zez'] == ['az', 'zczd', 'fzg']
+Text.split : Text | Vector Text  -> Case_Sensitivity -> Boolean -> Vector Text | Illegal_Argument
+Text.split self delimiter="," case_sensitivity=Case_Sensitivity.Sensitive use_regex=False =
+    delimiter_is_empty = case delimiter of
+        _ : Text -> delimiter.is_empty
+        _ : Vector -> delimiter.is_empty || delimiter.any (.is_empty)
+        _ -> Error.throw (Illegal_Argument.Error "The delimiter must be a Text or Vector of Texts.")
+    delimiter_is_singleton_vector = case delimiter of
+        _ : Vector -> delimiter.length == 1
+        _ -> False
+    if delimiter_is_empty then Error.throw (Illegal_Argument.Error "The delimiter cannot be empty or contain empty entries.") else
+        # If it's a vector of one element, just call it on that one element.
+        if delimiter_is_singleton_vector then self.split delimiter=(delimiter.at 0) case_sensitivity=case_sensitivity use_regex=use_regex else
+            case use_regex of
+                False ->
+                    delimiters = Vector.from_polyglot_array <|
+                        case delimiter of
+                            _ : Text -> case case_sensitivity of
+                                Case_Sensitivity.Sensitive ->
+                                    Text_Utils.span_of_all self delimiter
+                                Case_Sensitivity.Insensitive locale ->
+                                    Text_Utils.span_of_all_case_insensitive self delimiter locale.java_locale
+                            _ : Vector -> case case_sensitivity of
+                                Case_Sensitivity.Sensitive ->
+                                    Text_Utils.span_of_all_multiple self delimiter
+                                Case_Sensitivity.Insensitive locale ->
+                                    Text_Utils.span_of_all_case_insensitive_multiple self delimiter locale.java_locale
```

Comment:
If we renamed `span_of_all_multiple` to `span_of_all` I think the polyglot invocation would mean we could do just:

```suggestion
                        case case_sensitivity of
                                Case_Sensitivity.Sensitive ->
                                    Text_Utils.span_of_all self delimiter
                                Case_Sensitivity.Insensitive locale ->
                                    Text_Utils.span_of_all_case_insensitive self delimiter locale.java_locale
```

Reply:
Polyglot method overloading is very funky. And also IIRC it adds more overhead to the calls if there are overloads (but that would be probably a question to @JaroslavTulach). Anyway I know he advised me to not rely on Polyglot overload calls if I don't have to. So I think it's better to keep the method variants have different names.

Reply:
I agree with Radek here; it would work but it's a bit of magic and it wouldn't be obvious that it was happening without careful reading.